.github/workflows/ci.yml

```diff
@@ -0,0 +1,74 @@
+name: ci
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: ['*']
+
+jobs:
+  what-changed:
+    runs-on: ubuntu-latest
+    outputs:
+      packages: ${{ steps.set-packages.outputs.packages }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+      - name: Get changed packages
+        id: set-packages
+        run: |
+          exclude=(
+            "hono-middleware"
+            "@hono/bun-transpiler"
+          );
+          changed=$(yarn workspaces list --json --since | jq -nc --arg exclude "$exclude" '[inputs.name | select(inside($exclude) | not) | sub("@hono/"; "")]')
+          echo "packages=${changed}" >> $GITHUB_OUTPUT
+
+  build:
+    runs-on: ubuntu-latest
+    needs: [what-changed]
+    if: ${{ needs.what-changed.outputs.packages != '[]' }}
+    strategy:
+      matrix:
+        package: ${{ fromJSON(needs.what-changed.outputs.packages) }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+      - run: yarn workspaces focus hono-middleware @hono/${{ matrix.package }}
+      - run: yarn workspaces foreach --topological --recursive --from @hono/${{ matrix.package }} run build
+      - run: yarn workspace @hono/${{ matrix.package }} publint
+      - run: yarn workspace @hono/${{ matrix.package }} typecheck
+      - run: yarn eslint packages/${{ matrix.package }}
+      - run: yarn prettier --check . !packages packages/${{ matrix.package }}
```

Comment:
Uses the output from `what-changed` to run CI scripts on just the packages that have changes

The output from `what-changed` is used to build a dynamic [matrix strategy](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow), which is referenced by each step to run a script in a specific workspace, eg;

```sh
$ yarn workspace @hono/${{ matrix.package }} typecheck
``` 

..or, run a script using a specific workspace directory, eg;

```sh
$ yarn eslint packages/${{ matrix.package }}
```